### PR TITLE
[New model] Support CGNet

### PR DESCRIPTION
**Pull request by yamengxi**
Based on #107

### CODE REVIEW

**Comment by hellock** [comment]
Task linked: [CU-4huttc 支持CGNet](https://app.clickup.com/t/4huttc)

File: configs/_base_/models/cgnet_m3n21.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# model settings
```
Comment (xvjiarui) [inline configs/_base_/models/cgnet_m3n21.py:1]: We may rename this file to `cgnet.py`.

File: configs/cgnet/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+|-----------|----------|-----------|--------:|----------|----------------|------:|--------------:|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| CGNet | M3N21  | 680x680  |   60000 |      7.5 |           16.03 | 65.63 |     68.04 | [model](https://openmmlab.oss-accelerate.aliyuncs.com/mmsegmentation/v0.5/cgnet/cgnet_680x680_60k_cityscapes/cgnet_680x680_60k_cityscapes_20201101_110253-4c0b2f2d.pth) &#124; [log](https://openmmlab.oss-accelerate.aliyuncs.com/mmsegmentation/v0.5/cgnet/cgnet_680x680_60k_cityscapes/cgnet_680x680_60k_cityscapes-20201101_110253.log.json) |
```
Comment (xvjiarui) [inline configs/cgnet/README.md:20]: Use `https://download.openmmlab.com` instead.

File: mmseg/models/backbones/cgnet.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+@BACKBONES.register_module()
+class CGNet(nn.Module):
+    """CGNet backbone.
+
+    A Light-weight Context Guided Network for Semantic Segmentation
+    arXiv: https://arxiv.org/abs/1811.08201
+
+    Args:
+        in_channels (int): Number of input image channels. Normally 3.
+        num_channels (list[int]): Numbers of feature channels at each stages.
```
Comment (xvjiarui) [inline mmseg/models/backbones/cgnet.py:194]: Use tuple instead.

File: mmseg/models/decode_heads/fcn_head.py [hunk 4]
```diff
@@ -49,7 +49,10 @@ def __init__(self,
                     conv_cfg=self.conv_cfg,
                     norm_cfg=self.norm_cfg,
                     act_cfg=self.act_cfg))
-        self.convs = nn.Sequential(*convs)
+        if num_convs == 0:
+            self.convs = nn.Identity()
+        else:
+            self.convs = nn.Sequential(*convs)
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/fcn_head.py:55]: Add test for this change.

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**